epix/io.py

```diff
@@ -80,75 +81,106 @@ def loader(directory,
+        ttree, n_simulated_events = _get_ttree(directory, file_name)
+
+        if arg_debug:
+            print(f'Total entries in input file = {ttree.num_entries}')
+            cutby_string='output file entry'
+            if cut_by_eventid:
+                cutby_string='g4 eventid'
+
+            if kwargs['entry_start'] is not None:
+                print(f'Starting to read from {cutby_string} {kwargs["entry_start"]}')
+            if kwargs['entry_stop'] is not None:
+                print(f'Ending read in at {cutby_string} {kwargs["entry_stop"]}')
+
+        # If user specified entry start/stop we have to update number of
+        # events for source rate computation:
+        if kwargs['entry_start'] is not None:
+            start = kwargs['entry_start']
+        else:
+            start = 0
+
+        if kwargs['entry_stop'] is not None:
+            stop = kwargs['entry_stop']
+        else:
+            stop = n_simulated_events
+        n_simulated_events = stop - start
+
+        if cut_by_eventid:
+            # Start/stop refers to eventid so drop start drop from kwargs
+            # dict if specified, otherwise we cut again on rows.
+            kwargs.pop('entry_start', None)
+            kwargs.pop('entry_stop', None)
+
+        # Columns to be read from the root_file:
+        column_names = ["x", "y", "z", "t", "ed",
+                        "type", "trackid",
+                        "parenttype", "parentid",
+                        "creaproc", "edproc"]
+
+        # Conversions and parameters to be computed:
+        alias = {'x': 'xp/10',  # converting "geant4" mm to "straxen" cm
+                'y': 'yp/10',
+                'z': 'zp/10',
+                'r': 'sqrt(x**2 + y**2)',
+                't': 'time*10**9'
+                }
+
+        # Read in data, convert mm to cm and perform a first cut if specified:
+        interactions = ttree.arrays(column_names,
+                                    cut_string,
+                                    aliases=alias,
+                                    **kwargs)
+        eventids = ttree.arrays('eventid', **kwargs)
+        eventids = ak.broadcast_arrays(eventids['eventid'], interactions['x'])[0]
+        interactions['evtid'] = eventids
+
+    elif file.endswith(".csv"):
+
+        print("Load instructions from a csv file!")
+
+        instr_df =  pd.read_csv(file)
+        #unit conversion similar to root case
+        instr_df["x"] = instr_df["xp"]/10 
+        instr_df["y"] = instr_df["yp"]/10 
+        instr_df["z"] = instr_df["zp"]/10 
+        instr_df["r"] = np.sqrt(instr_df["x"]**2 + instr_df["y"]**2)
+        instr_df["t"] = instr_df["time"]*10**9
+
+        n_simulated_events = len(np.unique(instr_df.eventid))
+
+        if outer_cylinder:
+            instr_df = instr_df.query(cut_string)
+
+        interactions = awkwardify_df(instr_df)
+
+        #Use always all events in the csv file
+        start = 0
+        stop = n_simulated_events
```

Comment:
Is there a use case where we only want to read a part of the `csv` instructions? I guess no, but if so we need to add some sort of cut.